neuralmonkey/attention/base_attention.py

```diff
@@ -106,6 +109,10 @@ def __init__(self,
         with self.use_scope():
             self.train_mode = tf.placeholder(tf.bool, [], "train_mode")
 
+    @tensor
+    def batch_size(self) -> tf.Tensor:
+        raise NotImplementedError("Abstract property")
+
```

Comment:
Tohle tu musí být proč? Nikde v týhle třídě se `self.batch_size` nepoužívá, tak bych to nevynucoval.

Reply:
už vim proč to tu musí bejt. hierarchickej by nefungoval. disregard these comments.